### DISCUSSION TITLE
Privacy and self-hosted cloud editor

### DISCUSSION THREAD

**Discussion by Redsandro** [osmandapp/OsmAnd#18767] [Ideas]
1. There is this easy method to edit your favorites on `osmand.net/map` now, and its functionality is very useful. At the same time:
2. There are a quadrillion examples of personal data being used, abused, leaked and sold out there. There is a small group of people that avoids using public cloud infrastructure to store highly private data such as locations frequented and tracks recorded.

Is this something that can be solved? I would love to either:

1. Be able to self-host the web-based cloud favorite editor on my PC, NAS or VPS, similar to Matrix/Element where the app defaults to the official Matrix server but allows advanced options for entering your own server address; or
2. Have an option for OsmAnd to start a webserver so I can directly edit my favorites on the phone from my laptop, slightly similar to how Kodi can launch a webserver to control the app over http.

**Comment by anaxonda** [comment]
Great ideas.

**Comment by robinmetral** [comment]
And/or the option to save favorites to a cloud provider of a user's choice. Unfortunately it looks like the OsmAnd team used the cloud sync feature as a way to decline interoperability requests: https://github.com/osmandapp/OsmAnd/issues/2750#issuecomment-1785212964
